From 9732f4e80f906fab85b97ae55ee44bfd3ee4945e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Tue, 23 Feb 2021 23:42:11 +0000
Subject: meson: use encoding=utf-8 when reading/writing files in helper script

Fixes errors such as

Traceback (most recent call last):
  File "C:\Users\...\cairo\test\make-cairo-test-constructors.py", line 19, in <module>
    for l in f.readlines():
  File "c:\python39\lib\encodings\cp1253.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 6694: character maps to <undefined>

on non-English-language Windows locales/installations.
---
 boilerplate/make-cairo-boilerplate-constructors.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

(limited to 'boilerplate/make-cairo-boilerplate-constructors.py')

diff --git a/boilerplate/make-cairo-boilerplate-constructors.py b/boilerplate/make-cairo-boilerplate-constructors.py
index 8e02fa500..40198ed1a 100644
--- a/boilerplate/make-cairo-boilerplate-constructors.py
+++ b/boilerplate/make-cairo-boilerplate-constructors.py
@@ -16,12 +16,12 @@ if __name__=='__main__':
     match_boilerplate_name = re.compile(r'^CAIRO_BOILERPLATE.*\((.*),.*')
 
     for fname in args.input:
-        with open(fname, 'r') as f:
+        with open(fname, 'r', encoding='utf-8') as f:
             for l in f.readlines():
                 if match_boilerplate_line.match(l):
                     boilerplate_names.append(match_boilerplate_name.match(l).group(1))
 
-    with open(args.output, 'w') as f:
+    with open(args.output, 'w', encoding='utf-8') as f:
         f.write('/* WARNING: Autogenerated file - see %s! */\n\n' % sys.argv[0])
         f.write('#include "cairo-boilerplate-private.h"\n\n')
         f.write('void _cairo_boilerplate_register_all (void);\n\n')
